### PR TITLE
Replace read.csv with data.table::fread

### LINKED ISSUE #4: Replace read.csv with fread

### PR DESCRIPTION
**Pull request by adamobeng**
Fixes #4

### CODE REVIEW

**Comment by kbenoit** [comment]
I reviewed this, and while it works, it still relies on `readLines` to bring in the text as a `character` and then sends that to `fread`. The advantage of `fread` is that it offers a really fast method of going straight from the input file to the object. (Did you keep the `readLines` because of its encoding options through the connection?) How about doing this in `fread`directly?

Also you don't need to coerce the object, you can set the `data.table = FALSE` argument to have `fread` return a data.table.

**Comment by adamobeng** [comment]
I did keep `readLines` because it takes encoding options. We can't pass the encoding parameter to `fread` because it only supports Latin-1 and UTF-8 (and it seems to be [pretty hard-coded](https://github.com/Rdatatable/data.table/blob/b78fea5e513e323b606cae4e4e6b9a35163497cf/src/fread.c)). We also can't set the encoding on the connection because `fread` doesn't read from connections. 

We could use fread for those two encodings and read.csv for everything else, but that seems inelegant.

> Also you don't need to coerce the object, you can set the data.table = FALSE argument to have fread return a data.table.

Fixed.

**Comment by kbenoit** [comment]
What if we read it in with `fread`, and then `stri_conv` to and from after reading? In fact, the best way to make the conversions portable might be to do just that, read it in "raw" and then convert. The only problem will be if fread does some conversion of its own, thinking that something in an 8-bit encoding is UTF-8 and thereby mangling in upon import. But my guess is that it's just reading it in and attaching the Encoding bit based on what it finds. But we need to test.

**Comment by adamobeng** [comment]
I don't think that's possible. `fread` needs to know an encoding to do things like split the input into columns based on the delimiter. I can't even get it to read a UTF-16 file properly.

**Comment by kbenoit** [comment]
Hmmm, then we probably need to set up a processing workflow in which the encoding conversion etc is all handled the same way, before being processed by appropriate functions. The use of text connections as in 

```r
txt <- paste(readLines(con <- file(path), encoding=args$encoding, warn = FALSE), collapse="\n")
close(con)
```
and then sending to the appropriate processing function, such as `fread`, an XML handler, etc, is probably the most consistent method. As an alternative, if the files did NOT require conversion, we might be able to go directly to `fread`.

**Comment by adamobeng** [comment]
The version at @6200fcc uses fread directly for UTF-8 and Latin-1, and indirectly for all other encodings. 

The problem with handling encoding conversion in the same way is that the different packages used for different document types might support different encodings natively.

**Comment by kbenoit** [comment]
@adamobeng if you want to accept the PRs yourself, while you iron out the kinks in the list, I'm fine with that since it's just us two so far. But happy to do a review if you want a second pair of eyes too.

File: R/get_functions.R [hunk 1]
```diff
@@ -8,7 +8,28 @@ get_txt <- function(f, ...) {
...
+
+    # Replace native.enc with UTF-8 if that's what it is
+    if (args$encoding == 'native.enc') {  
+        #  http://r.789695.n4.nabble.com/Find-out-what-quot-native-enc-quot-corresponds-to-td4639208.html
+        args$encoding = strsplit(Sys.getlocale("LC_CTYPE"), '\\.')[[1]][2]
+    }
+    if (!(args$encoding %in% c('Latin-1', 'UTF-8'))) { 
+        # If the encoding is not one fread supports, open the file using R's native function
+        #  Use the encoding arg to open the file, pass all other args to fread
+        txt <- paste(readLines(con <- file(path), encoding=args$encoding, warn = FALSE), collapse="\n")
```
Comment (kbenoit) [inline R/get_functions.R]: I prefer 

```r
args <- c(list(input = txt, data.table = FALSE, stringsAsFactors = FALSE), args)
```

see https://github.com/kbenoit/quanteda/wiki/Style-guide

**Pull request merged**